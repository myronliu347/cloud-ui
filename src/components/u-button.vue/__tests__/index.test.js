import { mount } from '@vue/test-utils'
import Button from '../index.vue'

describe('Button', () => {
  it('should render properly', () => {
    const wrapper = mount(Button, {
      propsData: { text: 'primary' }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should call listener', async () => {
    const onClick = jest.fn()
    const wrapper = mount(Button, {
      listeners: { click: onClick }
    });
    await wrapper.find('a').trigger('click')
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
