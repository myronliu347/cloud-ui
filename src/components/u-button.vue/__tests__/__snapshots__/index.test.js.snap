// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render properly 1`] = `
<a
  class="root"
  hovertype="underline"
  tabindex="0"
  target="_self"
  vusion-slot-name-edit="text"
>
  <!---->
   primary
</a>
`;
